### DISCUSSION TITLE
Modelling Static (PQ) generators

### DISCUSSION THREAD

**Discussion by DEUCE1957** [rte-france/Grid2Op#652] [General]
In converting the CIGRE MV benchmark to Grid2Op, one issue I came across is that the generators in the CIGRE MV benchmark are modelled as static generators in PandaPower. However, it does not seem Grid2OP has any good way to represent these? I tried representing them as negative loads (which works) however this is hacky and causes other problems (breaks flexibility and also need to include my generators in the load_charac.csv file). 

![image](https://github.com/user-attachments/assets/31e87b84-0f3a-4501-a936-c43f153976f7)

Is it safe to represent the static (PQ) generators as normal generators in grid2op without making the benchmark invalid / inaccurate to the original?

**Comment by BDonnot** [comment]
Hello,

Your are totally correct and to be even more realistic, the modeling should be dynamic.

On real grid, sometimes generators are PV and some other time, later in the day, they are PQ. Grid2op (because it's a simplification of the reality) does not allow that currently. 
The best way would be to code everything related to this "new feature".

I did not code that because pandapower does not allow it and it would make the `PandaPowerBackend` a bit tricky to code.

To "represent" the same physical phenomenon of "PQ" generators you can :
- use load, as you have suggested but it will "mess with" the flexibility
- use storage, but it will discard the reactive (Q) part that you can "add" in a load for example
- use generator, but it will again discard the reactive part and even add some reactive so that the voltage is consistent with its setpoint.

I hope that helps
